storage/tests/system.py

```diff
@@ -723,82 +726,207 @@ def test_third_level(self):
 
+    @classmethod
+    def setUpClass(cls):
         if (
             type(Config.CLIENT._credentials)
-            is not google.oauth2.service_account.credentials
+            is not google.oauth2.service_account.Credentials
         ):
-            self.skipTest("Signing tests requires a service account credential")
-
-        logo_path = self.FILES["logo"]["path"]
-        with open(logo_path, "rb") as file_obj:
-            self.LOCAL_FILE = file_obj.read()
+            cls.skipTest("Signing tests requires a service account credential")
 
-        blob = self.bucket.blob("LogoToSign.jpg")
-        blob.upload_from_string(self.LOCAL_FILE)
-        self.case_blobs_to_delete.append(blob)
+        bucket_name = "gcp-signing" + unique_resource_id()
+        cls.bucket = Config.CLIENT.create_bucket(bucket_name)
+        cls.blob = cls.bucket.blob("README.txt")
+        cls.blob.upload_from_string(cls.BLOB_CONTENT)
 
-    def tearDown(self):
-        errors = (exceptions.TooManyRequests, exceptions.ServiceUnavailable)
+    @classmethod
+    def tearDownClass(cls):
+        _empty_bucket(cls.bucket)
+        errors = (exceptions.Conflict, exceptions.TooManyRequests)
         retry = RetryErrors(errors, max_tries=6)
-        for blob in self.case_blobs_to_delete:
-            if blob.exists():
-                retry(blob.delete)()
+        retry(cls.bucket.delete)(force=True)
 
-    def test_create_signed_read_url(self):
-        blob = self.bucket.blob("LogoToSign.jpg")
-        expiration = int(time.time() + 10)
-        signed_url = blob.generate_signed_url(
-            expiration, method="GET", client=Config.CLIENT
+    @staticmethod
+    def _morph_expiration(version, expiration):
+        if expiration is not None:
+            return expiration
+
+        if version == "v2":
+            return int(time.time()) + 10
+
+        return 10
+
+    def _create_signed_list_blobs_url_helper(
+        self, version, expiration=None, method="GET"
+    ):
+        expiration = self._morph_expiration(version, expiration)
+
+        signed_url = self.bucket.generate_signed_url(
+            expiration=expiration, method=method, client=Config.CLIENT, version=version
         )
 
         response = requests.get(signed_url)
         self.assertEqual(response.status_code, 200)
-        self.assertEqual(response.content, self.LOCAL_FILE)
 
-    def test_create_signed_read_url_lowercase_method(self):
-        blob = self.bucket.blob("LogoToSign.jpg")
-        expiration = int(time.time() + 10)
+    def test_create_signed_list_blobs_url_v2(self):
+        self._create_signed_list_blobs_url_helper(version="v2")
+
+    def test_create_signed_list_blobs_url_v2_w_expiration(self):
+        now = datetime.datetime.utcnow()
+        delta = datetime.timedelta(seconds=10)
+
+        self._create_signed_list_blobs_url_helper(expiration=now + delta, version="v2")
+
+    @pytest.mark.skip(reason="Back-end case-flattening bug: revisit 2019-04-03")
```

Comment:
You can unskip this test now and the one below.

Reply:
8f2e13f